docs/src/design/ten-thousand-foot-view.md

```diff
@@ -0,0 +1,211 @@
+observer* is responsible for handling events for map configuration
+changes. This includes events for loading and updating *style*, events
+for map becoming idle, events initialization and completion of
+rendering, and events for camera orientation changes. On the other hand,
+a rendering observer deals with events that are directly related to
+frame-by-frame map rendering. A rendering observer might publish a
+rendering event to map observers but they are usually rendering
+initialization or completion events. One key thing to remember here is a
+map configuration can lead to rendering changes such as camera switch,
+map center change, viewport size change, zoom, and pitch change.
+
+### Style
+
+Style component embodies a *MapLibre Style Document*. A style document
+is a JSON object that defines the visual appearance of a map through the
+order to draw it in, what to draw, and how to style the data while
+drawing it. Style sub-component is composed of classes and functions
+that allows downloading, and configuring a style. A style can be fetched
+from a remote HTTP resource or configured locally. The style component
+also carries a *Sprite Loader* to load sprites, remote HTTP URL for
+glyphs, layers to draw and sources of data to draw. A *sprite* is a
+single image that contains all the *icons* included in a style. Layers
+are composed of sources where a source could be a vector tile set,
+raster tile set, or GeoJSON data
+
+*To complete initialization of a map component, we need to initialize a
+map, style with source and layers, and observers.*
+
+### Layer
+
+*Layer* is an overloaded terminology in the MapLibre GL Native's
+context. *Layer* can mean any of the following:
+
+1.  From the point of view of data that needs to be rendered on map,
+    each map tile is composed of *layers* of data. Each layer, in this
+    context, contains *features*[^4] to be rendered in a map. These
+    features are defined by a *source.* Each *layer* is tied to a
+    *source*.
+
+2.  From the point of view of *style*, a style's layer property list all
+    the *layers* available in that *style*. A single style sheet can be
+    applied atop one or many layers. This layer definition converges
+    data to be rendered and the style to be applied on said layer.
+
+When this document uses the word layer in context of rendering, it
+refers to the definition mentioned in 2.
+
+### Glyphs
+
+A *glyph* is a single representation of a character. A font is a map of
+characters[^5]. Map tiles use labels of text to show name of cities,
+administrative boundaries, or street names. Map tiles also need to show
+icons for amenities like bus stops and parks. A map style uses character
+map from fonts to display labels and icons. Collectively these are
+called *glyphs.* A glyph is loaded as a set of signed-distance-fields in
+protobuf format. For curious readers, a signed-distance-field which
+takes a position and desired glyph as input and returns the distance
```

Comment:
I'm not sure your definition of SDFs is correct. The PBF contains a glyph _atlas_ (or glyphs). Each element of the atlas is a texture representing the SDF of the character to render.

A SDF uses a field of floats (signed-distance) to represent how a character (or anything else) should be drawn by the GPU. This has the advantage of scaling without artifacts. https://blog.mapbox.com/drawing-text-with-signed-distance-fields-in-mapbox-gl-b0933af6f817 would be the canonical link to explain this.

Reply:
I oversimplied this one. I deliberately did not talk about the Atlas here. I am reading about that part now so I can document it in a coming PR. 😅

I will read the link you shared and update this section. :)

Reply:
Thank you again. I definitely read it wrong. In my head it was about signed distance function. I have another task to add details about text rendering in this document. Probably after I understand the text rendering better.

Reply:
I updated the text to reflect what I learned. Considering this resolved. 😃